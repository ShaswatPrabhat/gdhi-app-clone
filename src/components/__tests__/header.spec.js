import { shallowMount, mount, RouterLinkStub } from "@vue/test-utils";
import { describe, expect, it, vi } from "vitest";
import Header from "../header/header.vue";
import { i18n } from "../../plugins/i18n";
import axios from "axios";

describe("Header ", () => {
  let wrapper;

  let countryData = [
    { id: "IND", name: "India", countryAlpha2Code: "IN" },
    { id: "USA", name: "United States of America", countryAlpha2Code: "US" },
    { id: "POL", name: "Poland", countryAlpha2Code: "PL" },
    { id: "AUS", name: "Australia", countryAlpha2Code: "AU" },
  ];
  const axiosGetSpy = vi.spyOn(axios, "get");
  axiosGetSpy.mockResolvedValue({ data: countryData });

  it("should have the data", () => {
    wrapper = shallowMount(Header, { i18n });
    expect(wrapper.vm.countries).to.deep.equal({});
    expect(wrapper.vm.developmentIndicators).to.deep.equal([]);
    expect(wrapper.vm.healthIndicators).to.deep.equal({});
  });

  it("should have the data", () => {
    wrapper = mount(Header, {
      stubs: {
        "router-link": RouterLinkStub,
      },
      i18n,
    });
    expect(wrapper.findAll(".hd-element").length).to.equal(4);
  });
});
