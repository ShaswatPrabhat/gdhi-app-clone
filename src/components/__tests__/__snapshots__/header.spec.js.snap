// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header  > should render header components 1`] = `
<div
  class="header"
  data-v-a605b683=""
>
  <div
    class="header-width"
    data-v-a605b683=""
  >
    <div
      class="header-elements"
      data-v-a605b683=""
    >
      <div
        class="left-aligned-headers"
        data-v-a605b683=""
      >
        <span
          class="logo-name hd-element"
          data-v-a605b683=""
        >
          <a
            data-v-a605b683=""
            href="http://www.digitalhealthmonitor.org"
            target="_blank"
          >
            <img
              alt="GDHM Logo"
              data-v-a605b683=""
              height="45px"
              src="/static/img/GDHMLogo.svg"
              width="200px"
            />
          </a>
        </span>
        <router-link
          class="hd-element header-link"
          data-v-a605b683=""
          to="[object Object]"
        >
          <span
            data-v-a605b683=""
          >
            World Map
          </span>
        </router-link>
        <a
          class="hd-element header-link"
          data-v-a605b683=""
        >
          <div
            class="dropdown"
            data-v-a605b683=""
          >
            <button
              class="dropbtn"
              data-v-a605b683=""
            >
               Regions 
              <span
                class="dropDown"
                data-v-a605b683=""
              />
            </button>
            <div
              class="dropdown-content-hide"
              data-v-a605b683=""
            >
              <a
                data-v-a605b683=""
                value="PAHO"
              >
                Pan American Region
              </a>
              <a
                data-v-a605b683=""
                value="AFRO"
              >
                African Region
              </a>
              <a
                data-v-a605b683=""
                value="EURO"
              >
                European Region
              </a>
              <a
                data-v-a605b683=""
                value="WPRO"
              >
                Western Pacific Region
              </a>
              <a
                data-v-a605b683=""
                value="SEARO"
              >
                South-East Asian Region
              </a>
              <a
                data-v-a605b683=""
                value="EMRO"
              >
                Eastern Mediterranean Region
              </a>
            </div>
          </div>
        </a>
        <router-link
          class="hd-element header-link"
          data-v-a605b683=""
          to="[object Object]"
        >
          <span
            data-v-a605b683=""
            style="margin: 0px 20px;"
          >
            Countries
          </span>
        </router-link>
        <router-link
          class="hd-element header-link"
          data-v-a605b683=""
          to="[object Object]"
        >
          <span
            data-v-a605b683=""
          >
            Indicators
          </span>
        </router-link>
        <router-link
          class="hd-element header-link"
          data-v-a605b683=""
          to="[object Object]"
        >
          <span
            data-v-a605b683=""
          >
            Methodology
          </span>
        </router-link>
      </div>
      <div
        class="header-language-selector"
        data-v-a605b683=""
      >
        <router-view
          data-v-a605b683=""
          name="languageSelect"
        />
      </div>
    </div>
  </div>
  <div
    class="sub-header"
    data-v-a605b683=""
  >
    <div
      class="sub-header-title-and-description"
      data-v-a605b683=""
    >
      <p
        class="sub-header-title"
        data-v-a605b683=""
      >
         State of Digital Health around the world today 
      </p>
      <p
        class="sub-header-description"
        data-v-a605b683=""
      >
         The Global Digital Health Monitor (GDHM) is an interactive web-based resource that aims to track, monitor, and
      assess the enabling environment for digital health throughout the world. 
      </p>
    </div>
    <div
      class="sub-header-map-legend"
      data-v-a605b683=""
    >
      <maplegend-stub
        data-v-a605b683=""
      />
    </div>
  </div>
</div>
`;
